### PR TITLE
Add convenience methods to read 3FGL

### PR DESCRIPTION
**Pull request by joleroi**
This PR introduces

* [x] Integral and differential flux points Table subclasses
* [x] Conversion method from int to diff (wrapper around existing functionality)
* [x] read_3FGL methods for both flux points classes as well as SpectrumFitResult
* [x] evaluate and evaluate_butterfly for SpectrumFitResult

I replaces https://github.com/gammapy/gammapy/pull/504

Up next
* The same for 2FHL (this is  a little bit more involved)

### CODE REVIEW

**Comment by joleroi** [comment]
I will merge this if tests pass - if you want to comment, hurry up :wink:

File: gammapy/image/maps.py [hunk 1]
```diff
@@ -342,13 +342,10 @@ def reproject(self, refheader, mode='interp', *args, **kwargs):
...
-            input_data=self.to_image_hdu(),
-            output_projection=refheader,
-            *args,
-            **kwargs,
-        )
-        map = SkyMap(name=self.name, data=out[0], wcs=self.wcs, unit=self.unit, meta=self.meta)
+        out = reproject_interp(input_data=self.to_image_hdu(),
+                               output_projection=refheader, *args, **kwargs)
+        map = SkyMap(name=self.name, data=out[0], wcs=self.wcs, unit=self.unit,
+                     meta=self.meta)
```
Comment (cdeil) [inline gammapy/image/maps.py]: This is an old change, no?
Why does it appear again in this PR?

File: gammapy/spectrum/results.py [hunk 2]
```diff
@@ -146,6 +148,37 @@ def __init__(self, spectral_model, parameters, parameter_errors,
...
+            parameters['alpha'] = Quantity(d['Spectral_Index'], '')
+            parameter_errors['alpha'] = Quantity(d['Unc_Spectral_Index'], '')
+            parameters['beta'] = Quantity(d['beta'], '')
+            parameter_errors['beta'] = Quantity(d['Unc_beta'], '')
+            parameters['reference'] = Quantity(d['Pivot_Energy'], 'MeV')
+            parameter_errors['reference'] = Quantity(0, 'MeV')
+            parameters['norm'] = Quantity(d['Flux_Density'], 'cm-2 s-1 MeV-1')
+            parameter_errors['norm'] = Quantity(d['Unc_Flux_Density'],
+                                                'cm-2 s-1 MeV-1')
+        else:
```
Comment (cdeil) [inline gammapy/spectrum/results.py]: If you have time, run `unique` on that column and add a TODO comment here which are the missing spectral models.
This kind of stuff is a good task for people to get started, but it has to be clear what exactly needs to be done.

File: gammapy/spectrum/results.py [hunk 3]
```diff
@@ -355,161 +392,175 @@ def to_sherpa_model(self, name='default'):
 
         return model
 
-    def plot(self, ax=None, energy_unit='TeV',
-             flux_unit='cm-2 s-1 TeV-1', e_power=0, **kwargs):
-        """Plot fit function
+    def evaluate(self, x):
+        """Wrapper around astropy.modelling.models.evaluate
```
Comment (cdeil) [inline gammapy/spectrum/results.py]: I'd suggest this:
"""Wrapper around the ``evaluate`` method on the Astropy model classes.
"""

File: gammapy/image/maps.py (hunk 1 above)
Comment (joleroi) [inline gammapy/image/maps.py]: Because I cherry picked the commit, to be able to run tests

File: gammapy/spectrum/results.py [hunk 4]
```diff
@@ -355,161 +392,175 @@ def to_sherpa_model(self, name='default'):
...
+        ----------
+        x : `~gammapy.utils.energy.Energy`
+            Evaluation point
+        """
+        if self.spectral_model == 'PowerLaw':
+            flux = models.PowerLaw1D.evaluate(x, self.parameters.norm,
+                                              self.parameters.reference,
+                                              self.parameters.index)
+        elif self.spectral_model == 'LogParabola':
+            #LogParabola evaluation does not work with arrays
```
Comment (cdeil) [inline gammapy/spectrum/results.py]: What!?

Can you post a minimal code snippet here that shows that it doesn't work?

Leave a TODO here or directly file an issue or PR with Astropy?

**Comment by cdeil** [comment]
Maybe add a test that exercises the `plot_butterfly` method?
(no asserts for now, just to make sure it runs fine for all builds)

**Comment by cdeil** [comment]
Ha ha ... I did manage to comment here before it went in. :smirk_cat:

File: gammapy/image/maps.py (hunk 1 above)
Comment (cdeil) [inline gammapy/image/maps.py]: Hmmm, :cherries:

File: gammapy/spectrum/results.py (hunk 2 above)
Comment (joleroi) [inline gammapy/spectrum/results.py]: I don't understand, sorry. What does this have to do with
http://docs.astropy.org/en/stable/api/astropy.table.unique.html 
I am not using a table here, am I?

File: gammapy/spectrum/results.py (hunk 4 above)
Comment (joleroi) [inline gammapy/spectrum/results.py]: I added a comment
see https://github.com/astropy/astropy/issues/4764

File: gammapy/image/maps.py (hunk 1 above)
Comment (joleroi) [inline gammapy/image/maps.py]: :yum:

**Comment by joleroi** [comment]
Yes the test situation with the results module is horrible. I will add some test in a follow-up PR (already in progress)

File: gammapy/spectrum/results.py (hunk 2 above)
Comment (cdeil) [inline gammapy/spectrum/results.py]: I meant call unique on the Fermi catalog flux model column to get a complete list of all models there are, so that it's clear what further work is needed.
Not call unique here in the code.
